### PR TITLE
Jsp and controllers

### PR DESCRIPTION
**Pull request by Victoria-Moiseienko**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/products/addProduct.jsp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<%--
```
Comment (jah701) [inline src/main/webapp/WEB-INF/views/products/addProduct.jsp]: Remove comments

**Review by jah701 — APPROVED** [review]
Мне не к чему придраться 👍

File: src/main/java/com/internet/shop/controllers/AddProductController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AddProductController extends HttpServlet {
+    @Inject
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    ProductService productService = (ProductService) injector.getInstance(ProductService.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/AddProductController.java]: what about access modifier?

File: src/main/java/com/internet/shop/controllers/AddProductController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/products/addProduct.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String productName = req.getParameter("productname");
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/AddProductController.java]:
```suggestion
        String productName = req.getParameter("productName");
```

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    private UserService userService = (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(req.getParameter("userId"));
+        String deletedUserName = userService.get(id).getName();
+        userService.delete(id);
+        List<User> allUsers = userService.getAll();
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/DeleteUserController.java]: let's remove his shopping cart as well

File: src/main/java/com/internet/shop/controllers/RegistrationController.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationController extends HttpServlet {
+    @Inject
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    UserService userService = (UserService) injector.getInstance(UserService.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/RegistrationController.java]: access modifiers

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShoppingCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+
+    @Inject
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    ShoppingCartService shoppingCartService =
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShoppingCartController.java]: same

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShoppingCartController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShoppingCartController.java]:
```suggestion
public class GetCartProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/ShowAllProductController.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShowAllProductController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShowAllProductController.java]:
```suggestion
public class ShowAllProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/ShowAllProductController.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    private static final Long USER_ID = 1L;
+
+    @Inject
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    ProductService productService = (ProductService) injector.getInstance(ProductService.class);
+    ShoppingCartService shoppingCartService =
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShowAllProductController.java]: same

File: src/main/java/com/internet/shop/controllers/ShowAllProductController.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        List<Product> allProducts = productService.getAll();
+
+        req.setAttribute("products", allProducts);
+        req.getRequestDispatcher("/WEB-INF/views/products/showProducts.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShowAllProductController.java]: do we need this method here? 
i can see that its logic is already implemented in AddProductToCartController?

File: src/main/java/com/internet/shop/controllers/UserController.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserController extends HttpServlet {
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/UserController.java]:
```suggestion
public class GetAllUsersController extends HttpServlet {
```

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -11,6 +11,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allUsers</servlet-name>
+        <servlet-class>com.internet.shop.controllers.UserController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allUsers</servlet-name>
+        <url-pattern>/users/allusers</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/users/all</url-pattern>
```

File: src/main/webapp/web.xml [hunk 13]
```diff
@@ -11,6 +11,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addproduct</servlet-name>
+        <servlet-class>com.internet.shop.controllers.AddProductController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addproduct</servlet-name>
+        <url-pattern>/products/addproduct</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/add</url-pattern>
```

File: src/main/webapp/web.xml [hunk 14]
```diff
@@ -11,6 +11,74 @@
...
+        <url-pattern>/users/delete</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProductToCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.AddProductToCartController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addProductToCart</servlet-name>
+        <url-pattern>/carts/addProduct</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml]: refactor url patterns according to example from checklist
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/com/internet/shop/controllers/AddProductController.java (hunk 2 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/AddProductController.java]: Done

File: src/main/java/com/internet/shop/controllers/AddProductController.java (hunk 3 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/AddProductController.java]: Done

File: src/main/java/com/internet/shop/controllers/DeleteUserController.java (hunk 4 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/DeleteUserController.java]: Done

File: src/main/java/com/internet/shop/controllers/RegistrationController.java (hunk 5 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/RegistrationController.java]: Done

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java (hunk 6 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShoppingCartController.java]: Done

File: src/main/java/com/internet/shop/controllers/ShoppingCartController.java (hunk 7 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShoppingCartController.java]: Done

File: src/main/java/com/internet/shop/controllers/ShowAllProductController.java (hunk 8 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShowAllProductController.java]: Done

File: src/main/java/com/internet/shop/controllers/ShowAllProductController.java (hunk 9 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShowAllProductController.java]: Done

File: src/main/java/com/internet/shop/controllers/UserController.java (hunk 11 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/UserController.java]: Done

File: src/main/webapp/web.xml (hunk 12 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 13 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 14 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml]: Done

File: src/main/java/com/internet/shop/controllers/ShowAllProductController.java (hunk 10 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShowAllProductController.java]: Done

File: src/main/webapp/web.xml [hunk 15]
```diff
@@ -11,6 +11,74 @@
...
+        <url-pattern>/users/delete</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProductToCart</servlet-name>
+        <servlet-class>com.internet.shop.controllers.AddProductToCartController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>addProductToCart</servlet-name>
+        <url-pattern>/carts/addproduct</url-pattern>
```
Comment (RomanDubovskyi) [inline src/main/webapp/web.xml:82]: fix according to example from checklist (under Good URLs naming)
https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/com/internet/shop/controllers/ShowAllProductsController.java [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ShowAllProductsController extends HttpServlet {
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShowAllProductsController.java]: remove empty line

File: src/main/java/com/internet/shop/controllers/ShowAllProductsController.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+    @Inject
+    private static Injector injector = Injector.getInstance("com.internet.shop");
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShowAllProductsController.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/java/com/internet/shop/controllers/ShowAllProductsController.java [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+    private static ProductService productService =
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/ShowAllProductsController.java]: why static?

File: src/main/java/com/internet/shop/controllers/RegistrationController.java [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+import com.internet.shop.service.ShoppingCartService;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationController extends HttpServlet {
+    @Inject
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/RegistrationController.java]: what does this annotation do here?

File: src/main/webapp/web.xml (hunk 15 above)
Comment (Victoria-Moiseienko) [reply src/main/webapp/web.xml:82]: Done

File: src/main/java/com/internet/shop/controllers/ShowAllProductsController.java (hunk 16 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShowAllProductsController.java]: Done

File: src/main/java/com/internet/shop/controllers/ShowAllProductsController.java (hunk 17 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShowAllProductsController.java]: Done

File: src/main/java/com/internet/shop/controllers/ShowAllProductsController.java (hunk 18 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/ShowAllProductsController.java]: Done

File: src/main/java/com/internet/shop/controllers/RegistrationController.java (hunk 19 above)
Comment (Victoria-Moiseienko) [reply src/main/java/com/internet/shop/controllers/RegistrationController.java]: Done

File: src/main/java/com/internet/shop/controllers/InsertDataController.java [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class InsertDataController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private static final Product product1 = new Product("apple", 25.5);
+    private static final Product product2 = new Product("pear", 35.5);
+    private static final Product product3 = new Product("plum", 20);
+    private static final User user1 = new User("Vasil", "fruitlover", "qwerty");
+    private static final User user2 = new User("Maria", "mariafruit", "asdfgh");
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/controllers/InsertDataController.java]: you can declare them as local variables

File: src/main/webapp/web.xml (hunk 15 above)
Comment (RomanDubovskyi) [reply src/main/webapp/web.xml:82]: let's use this example 
image

**Review by RomanDubovskyi — APPROVED** [review]
fix url mapping according to checklist and you can merge pr after

**Pull request merged**